api/src/Auth/Driver/Guard.php

```diff
@@ -0,0 +1,351 @@
+    public function __construct(
+        JWTInterface $tokenizer,
+        PayloadFactory $payloadFactory,
+        PayloadValidationFactory $validationFactory,
+        UserProvider $provider,
+        RequestParser $requestParser
+    )
+    {
+
+        $this->payloadFactory = $payloadFactory;
+        $this->provider = $provider;
+        $this->tokenizer = $tokenizer;
+        $this->validationFactory = $validationFactory;
+        $this->requestParser = $requestParser;
+    }
+
+    /**
+     * Determine if the current user is authenticated.
+     *
+     * @return bool
+     */
+    public function check()
+    {
+        return (bool)$this->user;
+    }
+
+    /**
+     * Determine if the current user is a guest.
+     *
+     * @return bool
+     */
+    public function guest()
+    {
+        return !$this->check();
+    }
+
+    /**
+     * Get the currently authenticated user.
+     *
+     * @return \Illuminate\Contracts\Auth\Authenticatable|null
+     */
+    public function user()
+    {
+        if ($this->user === false){
+            return null;
+        }
+
+        if ($this->user){
+            return $this->user;
+        }
+
+        try{
+            $user = $this->getUserFromRequest();
+        }catch (\Exception $e){
+            return null;
+        }
+
+        if ($user){
+            $this->user = $user;
+            $this->viaToken = true;
+        }
+
+        return $user;
+    }
+
+    public function getUserFromRequest()
+    {
+        $token = $this->getRequestParser()->getToken($this->getRequest());
+        $payload = $this->getTokenizer()->decode($token);
+        $this->getValidationFactory()->validatePayload($payload);
+        $user = $this->getProvider()->retrieveByToken(null, $payload);
+        return $user;
+    }
+
+    /**
+     * Log a user into the application without token.
+     *
+     * @param  array $credentials
+     * @return bool
+     */
+    public function once(array $credentials = [])
+    {
+        return $this->attempt($credentials, false, true);
+    }
+
+    /**
+     * Attempt to authenticate a user using the given credentials.
+     *
+     * @param  array $credentials
+     * @param  bool $remember
+     * @param  bool $login
+     * @return bool
+     */
+    public function attempt(array $credentials = [], $remember = true, $login = true)
+    {
+        $user = $this->getProvider()->retrieveByCredentials($credentials);
+        if ($user  && $this->getProvider()->validateCredentials($user, $credentials)){
+            if ($login) {
+                $this->login($user, $remember);
+            }
+
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * Attempt to authenticate using HTTP Basic Auth.
+     *
+     * @param  string $field
+     * @return null|\Symfony\Component\HttpFoundation\Response
+     * @throws \Exception
+     */
+    public function basic($field = 'email')
+    {
+        throw new \Exception('Not implemented');
```

Comment:
It's in the other scope, so it would depend on App, I've built it as a package.
I can add some `src/Contracts/Exceptions` so this package in future be dependant on Spira/Contracts